cmds/core/dirname/dirname.go

```diff
@@ -9,17 +9,27 @@ package main
 
 import (
 	"fmt"
+	"io"
 	"log"
 	"os"
 	"path/filepath"
 )
 
-func main() {
-	if len(os.Args) < 2 {
-		log.Fatalf("dirname: missing operand")
+func runDirname(w io.Writer, args []string) error {
+	if len(args) < 1 {
+		fmt.Fprintf(w, "%s\n", "dirname: missing operand")
```

Comment:
functions should not print. Can you return this string as an error?

but again, if you used my suggested prototype, this becomes a compilation error. 

But then you'll need to test len(flag.Args()) in main, I get it, but still ... I prefer the required dirname argument.